packages/react/src/components/input/input.tsx

```diff
@@ -1,21 +1,46 @@
 import { c, classy, InputProps as BaseProps, m } from '@onfido/castor';
-import React from 'react';
+import React, { useState } from 'react';
+import { FieldLabelWrapper } from '../../internal';
 import { withRef } from '../../utils';
 
+const idPrefix = 'castor_input';
+let idCount = 0;
+
 export const Input = withRef(
   (
-    { type = 'text', invalid, className, ...restProps }: InputProps,
+    {
+      id: externalId,
+      type = 'text',
+      invalid,
+      children,
+      className,
+      ...restProps
+    }: InputProps,
     ref: InputProps['ref']
-  ): JSX.Element => (
-    <input
-      {...restProps}
-      ref={ref}
-      type={type}
-      className={classy(c('input'), m({ invalid }), className)}
-    />
-  )
+  ): JSX.Element => {
+    const [autoId] = useState<string | undefined>(() =>
+      children ? `${idPrefix}_${++idCount}` : undefined
```

Comment:
Yeah, you're right.

Thought have to use `children ? autoId : undefined` otherwise:

<img width="1099" alt="Screenshot 2021-02-09 at 15 50 07" src="https://user-images.githubusercontent.com/20243687/107389201-7e615200-6aee-11eb-835a-7cea80223e42.png">